fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/digitalcash/DigitalCashIssueFragment.java

```diff
@@ -36,175 +36,166 @@
+          .show();
     }
-
-    /**
-     * Function that return the give list of attendees Radio Group Button selected (a empty list if
-     * nothing)
-     */
-    private Set<PublicKey> attendeesPerRadioGroupButton(int radioGroup, String currentSelected) throws NoRollCallException {
-        Set<PublicKey> attendees = new HashSet<>();
-        if (radioGroup == selectOneMember && !currentSelected.equals("")) {
-            attendees.add(new PublicKey(currentSelected));
-        } else if (radioGroup == selectAllLaoMembers) {
-            for (RollCall current :
-                    Objects.requireNonNull(mViewModel.getCurrentLao()).getRollCalls().values()) {
-                attendees.addAll(current.getAttendees());
-            }
-        } else if (radioGroup == selectAllRollCallAttendees) {
-            attendees = mViewModel.getAttendeesFromTheRollCall();
-        } else if (radioGroup == selectAllLaoWitnesses) {
-            attendees = Objects.requireNonNull(mViewModel.getCurrentLao()).getWitnesses();
+  }
+
+  public Map<String, String> computeMapForPostTransaction(
+      String currentAmount, String currentPublicKeySelected, int radioGroup)
+      throws NoRollCallException {
+    if (radioGroup == DigitalCashViewModel.NOTHING_SELECTED) {
+      // In unlikely event that no radiobutton are selected, it do as if the first one was selected
+      return Collections.singletonMap(currentPublicKeySelected, currentAmount);
+    } else {
+      Set<PublicKey> attendees = attendeesPerRadioGroupButton(radioGroup, currentPublicKeySelected);
+      Map<String, String> issueMap = new HashMap<>();
+      if (!attendees.isEmpty()) {
+        for (PublicKey publicKey : attendees) {
+          issueMap.putIfAbsent(publicKey.getEncoded(), currentAmount);
         }
-        return attendees;
+      }
+      return issueMap;
     }
-
-    /**
-     * Function that setup the Button
-     */
-    private void setupSendCoinButton() {
-        mBinding.digitalCashIssueIssue.setOnClickListener(v -> mViewModel.postTransactionEvent());
+  }
+
+  /**
+   * Function that return the give list of attendees Radio Group Button selected (a empty list if
+   * nothing)
+   */
+  private Set<PublicKey> attendeesPerRadioGroupButton(int radioGroup, String currentSelected)
+      throws NoRollCallException {
+    Set<PublicKey> attendees = new HashSet<>();
+    if (radioGroup == selectOneMember && !currentSelected.equals("")) {
+      attendees.add(new PublicKey(currentSelected));
+    } else if (radioGroup == selectAllLaoMembers) {
+      for (RollCall current :
+          Objects.requireNonNull(mViewModel.getCurrentLaoValue()).getRollCalls().values()) {
+        attendees.addAll(current.getAttendees());
+      }
+    } else if (radioGroup == selectAllRollCallAttendees) {
+      attendees = mViewModel.getAttendeesFromTheRollCall();
+    } else if (radioGroup == selectAllLaoWitnesses) {
```

Comment:
I know `else if`s are a good precaution but just thinking out loud, could some of these conditions be true at the same time and thus we have a priority ranking and that's why we use `else if`s instead of `if`s

Reply:
I didn't thought about it, perjaps change the setting of the radiogroup to select only one button
But actually lao attendees contain rollcall attendees which may justify the first if else

Reply:
I did not change that so this is what's on master. My take on this is that's it's fine like that. We could change with `if` and return from each condition but I don't think it's better

Reply:
No no, it is more than fine like this I was just wondering if there was a specific order that we have to perform the checks but it is a good sanity check that it is if else instead of simple if

Reply:
Oh I misunderstood sorry. I think that the underlying UI component does not allow for multiple selection in this case. I think it is a slight optimization to use if/else if because the other condition won't have to be checked